pkg/reconciler/dependencybuild/buildrecipeyaml.go

```diff
@@ -1,541 +1,192 @@
+import (
+	_ "embed"
+	"github.com/redhat-appstudio/jvm-build-service/pkg/reconciler/artifactbuild"
+	"github.com/redhat-appstudio/jvm-build-service/pkg/reconciler/configmap"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+	pipelinev1beta1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"strings"
+)
 
-          curl --verbose --fail http://localhost:2000/deploy/result -o $(results.contaminants.path) || cat $(workspaces.maven-settings.path)/sidecar.log
-          cat $(workspaces.maven-settings.path)/sidecar.log
-        resources:
-          requests:
-            memory: "128Mi"
-            cpu: "10m"
-          limits:
-            memory: "512Mi"
-            cpu: "300m"
-`
+const (
+	WorkspaceBuildSettings = "build-settings"
+	WorkspaceSource        = "source"
 )
+
+//go:embed scripts/maven-settings.sh
+var mavenSettings string
+
+//go:embed scripts/gradle-settings.sh
+var gradleSettings string
+
+//go:embed scripts/maven-build.sh
+var mavenBuild string
+
+//go:embed scripts/gradle-build.sh
+var gradleBuild string
+
+//go:embed scripts/deploy.sh
+var deploy string
+
+func createPipelineSpec(maven bool, sidecarImage string, namespace string) *pipelinev1beta1.PipelineSpec {
+	var settings string
+	var build string
+	if maven {
+		settings = mavenSettings
+		build = mavenBuild
+	} else {
+		settings = gradleSettings
+		build = gradleBuild
+	}
+	zero := int64(0)
+	buildSetup := pipelinev1beta1.TaskSpec{
+		Results:    []pipelinev1beta1.TaskResult{{Name: PipelineResultContaminates, Description: "Any community GAVs that have ended up in the final output."}},
+		Workspaces: []pipelinev1beta1.WorkspaceDeclaration{{Name: WorkspaceBuildSettings}, {Name: WorkspaceSource}},
+		Params: []v1alpha1.ParamSpec{
+			{Name: PipelineScmUrl, Type: pipelinev1beta1.ParamTypeString},
+			{Name: PipelineScmTag, Type: pipelinev1beta1.ParamTypeString},
+			{Name: PipelineImage, Type: pipelinev1beta1.ParamTypeString},
+			{Name: PipelineGoals, Type: pipelinev1beta1.ParamTypeArray},
+			{Name: PipelineJavaVersion, Type: pipelinev1beta1.ParamTypeString},
+			{Name: PipelineToolVersion, Type: pipelinev1beta1.ParamTypeString},
+			{Name: PipelinePath, Type: pipelinev1beta1.ParamTypeString},
+			{Name: PipelineEnforceVersion, Type: pipelinev1beta1.ParamTypeString},
+			{Name: PipelineIgnoredArtifacts, Type: pipelinev1beta1.ParamTypeString},
+		},
+		Sidecars: []pipelinev1beta1.Sidecar{createSidecar(sidecarImage, namespace)},
+		Steps: []pipelinev1beta1.Step{
+			{
+				Container: v1.Container{
+					Name:  "git-clone",
+					Image: "gcr.io/tekton-releases/github.com/tektoncd/pipeline/cmd/git-init:v0.21.0", //TODO: should not be hard coded
+					Resources: v1.ResourceRequirements{
+						Requests: v1.ResourceList{"memory": resource.MustParse("128Mi"), "cpu": resource.MustParse("10m")},
```

Comment:
minimally add TODO comments around all these resource mem/cpu settings, unless you want to take it on in this PR

long term like the image ref todo's you have, these need to be configurable

Reply:
I have started this for the cache at https://github.com/redhat-appstudio/jvm-build-service/pull/233, am planning on going through and making all the limits configurable at some point.

Configuring the image is a bit trickier, it probably should be using the task from the build definitions bundle.

Reply:
sounds good

now, in #211 you bumped this from 128Mi to 256Mi; I know #211 was a bit of a guesstimate, but I just wanted to confirm what you think the default should be here

I'm also thinking we should move whatever changes are in #211 that we want to keep to this PR, and then close #211, right?